AmplifyPlugins/Logging/Tests/AWSCloudWatchLoggingPluginTests/LogRotationTests.swift

```diff
@@ -0,0 +1,157 @@
+            "amplify.0.log"
+        ])
+    }
+    
+    /// Given: a log rotation
+    /// When: a rotation occurs
+    /// Then: then a new rotated log file is created
+    func testLogRotationCreatesNewFiles() throws {
+        let originalContents = try FileManager.default.contentsOfDirectory(at: directory, includingPropertiesForKeys: nil)
+        XCTAssertEqual(originalContents.map { $0.lastPathComponent }, [
+            "amplify.0.log"
+        ])
+
+        systemUnderTest = try LogRotation(directory: directory,
+                                          fileSizeLimitInBytes: fileSizeLimitInBytes)
+        XCTAssertEqual(systemUnderTest.currentLogFile.fileURL.lastPathComponent, "amplify.1.log")
+        try systemUnderTest.rotate()
+        
+        var rotatedContents = try FileManager.default.contentsOfDirectory(at: directory, includingPropertiesForKeys: nil)
+        XCTAssertEqual(rotatedContents.map { $0.lastPathComponent }, [
+            "amplify.2.log",
+            "amplify.1.log",
+            "amplify.0.log",
+        ])
+        
+        try systemUnderTest.rotate()
+        rotatedContents = try FileManager.default.contentsOfDirectory(at: directory, includingPropertiesForKeys: nil)
+        XCTAssertEqual(rotatedContents.map { $0.lastPathComponent }, [
+            "amplify.2.log",
+            "amplify.3.log",
+            "amplify.1.log",
+            "amplify.0.log",
+        ])
+        
+        try systemUnderTest.rotate()
+        rotatedContents = try FileManager.default.contentsOfDirectory(at: directory, includingPropertiesForKeys: nil)
+        XCTAssertEqual(rotatedContents.map { $0.lastPathComponent }, [
+            "amplify.4.log",
+            "amplify.2.log",
+            "amplify.3.log",
+            "amplify.1.log",
+            "amplify.0.log",
+        ])
+        
+        systemUnderTest = try LogRotation(directory: directory,
+                                          fileSizeLimitInBytes: fileSizeLimitInBytes)
+        XCTAssertEqual(systemUnderTest.currentLogFile.fileURL.lastPathComponent, "amplify.0.log")
+    }
+    
+    /// Given: a log rotation
+    /// When: rotation occurs to the max limit
+    /// Then: the log rotation circles back to 0
+    func testLogRotationToMaxLimit() async throws {
+        for _ in 0..<(fileCountLimit) {
+            XCTAssertEqual(systemUnderTest.currentLogFile.available, systemUnderTest.currentLogFile.sizeLimitInBytes)
+            try systemUnderTest.rotate()
+            try await Task.sleep(seconds: 0.10)
+        }
+        let allFiles = try FileManager.default.contentsOfDirectory(at: directory, includingPropertiesForKeys: nil)
+        XCTAssertEqual(Set(allFiles.map { $0.lastPathComponent }), Set([
+            "amplify.0.log",
+            "amplify.1.log",
+            "amplify.2.log",
+            "amplify.3.log",
+            "amplify.4.log"
+        ]))
+    }
+    
+    func testLogRotationUseUnderutilizedLogFile() async throws {
```

Comment:
nit: given-when-then doc

Reply:
added missing give/when/then documentation to the rest of the tests